File: psets/pset-6/exercises/cash.py
from cs50 import get_float

cash = 0
total = 0

# GET INPUT
while cash <= 0:
    cash = get_float("Change owed: ")

# AVOID FLOAT PRECISION ISSUES
cash_in_cents = int(cash * 100)

# GET INTEGER PART OF DIVISION
total += cash_in_cents // 25
# GET INTEGER REST OF DIVISION
cash_in_cents %= 25

total += cash_in_cents // 10
cash_in_cents %= 10

total += cash_in_cents // 5
cash_in_cents %= 5

total += cash_in_cents

# PRINT TOTAL OF COINS
print(total)

File: psets/pset-1/exercises/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void) {
    // INPUT
    float input = 0.0;

    do {
        input = get_float("Cash: ");
    } while(input <= 0.0);

    // VARIABLES
    int total = 0, value = 0;

    // ROUND INPUT
    value = round(input * 100);

    // MAIN SECTION
    while(value > 0) {
        if(value >= 25) {
            total += 1;
            value -= 25;
        }

        if(value < 25 && value >= 10) {
            total += 1;
            value -= 10;
        }

        if(value < 10 && value >= 5) {
            total += 1;
            value -= 5;
        }

        if(value < 5 && value >= 1) {
            total += 1;
            value -= 1;
        }
    }

    // OUTPUT
    printf("%d\n", total);

    return 0;
}
